Panic when docker is not running at deploy time

## Issue

The CLI panics when building the images during `nitric up`. This is to do with docker not running which causes the nil pointer dereference. The error does not occur if docker is already running.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x10167e69c]

goroutine 102 [running]:
github.com/nitrictech/cli/pkg/build.buildFunction.func1()
	/Users/ryan/Documents/Code/nitric/cli/pkg/build/build.go:76 +0x21c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/ryan/go/pkg/mod/golang.org/x/sync@v0.2.0/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	/Users/ryan/go/pkg/mod/golang.org/x/sync@v0.2.0/errgroup/errgroup.go:72 +0xa4
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x10167e69c]

goroutine 101 [running]:
github.com/nitrictech/cli/pkg/build.buildFunction.func1()
	/Users/ryan/Documents/Code/nitric/cli/pkg/build/build.go:76 +0x21c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/ryan/go/pkg/mod/golang.org/x/sync@v0.2.0/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	/Users/ryan/go/pkg/mod/golang.org/x/sync@v0.2.0/errgroup/errgroup.go:72 +0xa4
```

## Expects

When docker is not running a helpful output message is displayed:

```
Nitric relies on Docker to be running before containerizing your project. Please refer to our installation instructions - https://nitric.io/docs/installation
```

--- a/pkg/build/build.go
+++ b/pkg/build/build.go
@@ -42,7 +42,10 @@ func buildFunction(s *project.Project, f project.Function) func() error {
 	fun := &f
 
 	return func() error {
-		ce, _ := containerengine.Discover()
+		ce, err := containerengine.Discover()
+		if err != nil {
+			return err
+		}
 
 		rt, err := fun.GetRuntime()
 		if err != nil {

